src/ESPAsyncWebServer.h

```diff
@@ -292,9 +290,11 @@ class AsyncWebServerRequest {
     bool hasParam(const String& name, bool post=false, bool file=false) const;
     bool hasParam(const __FlashStringHelper * data, bool post=false, bool file=false) const;
 
-    AsyncWebParameter* getParam(const String& name, bool post=false, bool file=false) const;
-    AsyncWebParameter* getParam(const __FlashStringHelper * data, bool post, bool file) const;
-    AsyncWebParameter* getParam(size_t num) const;
+    const AsyncWebParameter* getParam(const String& name, bool post=false, bool file=false) const;
+#ifdef ESP8266
+    const AsyncWebParameter* getParam(const __FlashStringHelper * data, bool post, bool file) const;
+#endif
+    const AsyncWebParameter* getParam(size_t num) const;
```

Comment:
Hmmm... that was one of the weak spots (among the others)...
you should declare it as `const AsyncWebParameter* p = request->getParam(i);`

the thing is that `getParam()` is declared as `getParam() const;`, so literally it should not return anything that could be changed outside of the object. I faced it when reworking the iterators.
So need to go in one way or the other - remove both `const`'s or add both. Adding `const` in return type could be  a breaking change yes. So, should we remove `const` from a method declaration also?
Logically it should return `const AsyncWebParameter*` since you are accessing params for read-only purposes usually, not supposed to change anything in request, right?
There is no "compatible" way here, this is a design flaw. I'm open to discussion :)

Reply:
> you should declare it as `const AsyncWebParameter* p = request->getParam(i);`

I know...

But the previous API **was not enforcing the use of** const, so making it const now will break all libraries and projects using it without const. So that's not an option. I will repeat myself again: **No API change**.

Maybe I will allow breaking the API in the future **once this fork will be largely adopted**, but for now, it has to stay API compatible in order to **facilitate its adoption** from people.

You could keep is as-is and for now use a const_cast. I was able to do it to test yesterday the changes in my app.